src/client.ts

```diff
@@ -222,26 +226,35 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
   #getServiceError(failure: QueryFailure, httpStatus: number): ServiceError {
     switch (httpStatus) {
       case 400:
+        if (queryCheckFailureCodes.includes(failure.error.code)) {
+          return new QueryCheckError(failure);
+        }
+        if (failure.error.code === "invalid_request") {
+          return new InvalidRequestError(failure);
+        }
         if (
-          httpStatus === 400 &&
-          queryCheckFailureCodes.includes(failure.error.code)
+          failure.error.code === "abort" &&
+          failure.error.abort !== undefined
         ) {
-          return new QueryCheckError(failure, httpStatus);
+          return new AbortError(
+            failure as QueryFailure & { error: { abort: QueryValue } }
+          );
         }
-
-        return new QueryRuntimeError(failure, httpStatus);
+        return new QueryRuntimeError(failure);
       case 401:
-        return new AuthenticationError(failure, httpStatus);
+        return new AuthenticationError(failure);
       case 403:
-        return new AuthorizationError(failure, httpStatus);
+        return new AuthorizationError(failure);
+      case 409:
+        return new ContendedTransactionError(failure);
       case 429:
-        return new ThrottlingError(failure, httpStatus);
+        return new ThrottlingError(failure);
       case 440:
-        return new QueryTimeoutError(failure, httpStatus);
+        return new QueryTimeoutError(failure);
       case 500:
-        return new ServiceInternalError(failure, httpStatus);
+        return new ServiceInternalError(failure);
       case 503:
-        return new ServiceTimeoutError(failure, httpStatus);
+        return new ServiceTimeoutError(failure);
```

Comment:
nit, but IMO this makes the code more brittle as now we're losing the type checking that we're passing the right error codes to the right errors.

Reply:
That makes sense. I changed it back.